rotkehlchen/chain/ethereum/modules/shutter/decoder.py

```diff
@@ -0,0 +1,123 @@
+from rotkehlchen.logging import RotkehlchenLogsAdapter
+from rotkehlchen.types import ChecksumEvmAddress
+from rotkehlchen.utils.misc import hex_or_bytes_to_address, hex_or_bytes_to_int
+
+from .constants import CPT_SHUTTER, REDEEMED_VESTING, SHUTTER_AIDROP_CONTRACT
+
+if TYPE_CHECKING:
+    from rotkehlchen.chain.evm.decoding.base import BaseDecoderTools
+    from rotkehlchen.chain.evm.node_inquirer import EvmNodeInquirer
+    from rotkehlchen.user_messages import MessagesAggregator
+
+logger = logging.getLogger(__name__)
+log = RotkehlchenLogsAdapter(logger)
+
+
+class ShutterDecoder(DecoderInterface):
+
+    def __init__(
+            self,
+            evm_inquirer: 'EvmNodeInquirer',
+            base_tools: 'BaseDecoderTools',
+            msg_aggregator: 'MessagesAggregator',
+    ) -> None:
+        super().__init__(
+            evm_inquirer=evm_inquirer,
+            base_tools=base_tools,
+            msg_aggregator=msg_aggregator,
+        )
+        self.shu = A_SHU.resolve_to_evm_token()
+
+    def _decode_shutter_claim(self, context: DecoderContext) -> DecodingOutput:
+        if not (
+            context.tx_log.topics[0] == REDEEMED_VESTING and
+            self.base.is_tracked(hex_or_bytes_to_address(context.tx_log.topics[2]))
+        ):
+            return DEFAULT_DECODING_OUTPUT
+
+        redeem_id = vesting_contract_address = None
+        for tx_log in reversed(context.all_logs):
+            if tx_log.topics[0] == REDEEMED_VESTING:
+                redeem_id = tx_log.topics[1]
+            elif (  # AddedVesting (id)
+                tx_log.topics[0] == b'\xdc\x17\xa5b\x85\x03\x17a\xfbz\x86\xd9\n1"u1 ^\x1e\x9e\x8f\x85\xf4\xd0\xf0:T\xea_\xb9v' and  # noqa: E501
+                tx_log.topics[1] == redeem_id
+            ):
+                vesting_contract_address = tx_log.address
+            elif (
+                tx_log.address == self.shu.evm_address and
+                tx_log.topics[0] == ERC20_OR_ERC721_TRANSFER and
+                hex_or_bytes_to_address(tx_log.topics[1]) == SHUTTER_AIDROP_CONTRACT and
+                hex_or_bytes_to_address(tx_log.topics[2]) == vesting_contract_address
+            ):
+                amount = token_normalized_value(token_amount=hex_or_bytes_to_int(tx_log.data), token=self.shu)  # noqa: E501
+                break
+        else:
+            log.error(f'Could not find the SHU transfer in {context.transaction.tx_hash.hex()}')
+            return DEFAULT_DECODING_OUTPUT
+
+        return DecodingOutput(event=self.base.make_event_from_transaction(
+            transaction=context.transaction,
+            tx_log=context.tx_log,
+            event_type=HistoryEventType.INFORMATIONAL,
+            event_subtype=HistoryEventSubType.NONE,
+            asset=self.shu,
+            balance=Balance(amount=FVal(amount)),
+            location_label=hex_or_bytes_to_address(context.tx_log.topics[2]),
```

Comment:
Why did you mark this as resolved. You are not using the user address from the variable as I said, but you are recalculating it.
```suggestion
            location_label=user_address,
```